Cython/Compiler/ExprNodes.py

```diff
@@ -2055,7 +2064,12 @@ def analyse_target_declaration(self, env):
             self.entry = env.lookup_here(self.name)
         if not self.entry and self.annotation is not None:
             # name : type = ...
-            self.declare_from_annotation(env, as_target=True)
+            if 'dataclasses.dataclass' in env.directives:
+                # in a dataclass an assignment should not prevent a name becoming a class member
```

Comment:
Did you mean "instance attribute" or "class attribute" here?
```suggestion
                # in a dataclass an assignment should not prevent a name becoming an instance attribute
```